Parsers/ASimNetworkSession/Parsers/vimNetworkSessionCrowdStrikeFalconHost.yaml

```diff
@@ -0,0 +1,379 @@
+      "2099200", "Prevention-blocked", "Deny", "Failure",
+      "2099328", "Prevention-blocked,quarantine", "Deny", "Failure",
+      "4196352", "Prevention-blocked", "Deny", "Failure",
+      "4196480", "Prevention-blocked,quarantine", "Deny", "Failure",
+      "1048576", "Prevention-suspend", "Deny", "Failure",
+      "524288", "Prevention-suspend", "Deny", "Failure",
+      "262144", "Blocking Disabled", "Allow", "Success",
+      "16384", "Safeguard Enabled", "Allow", "Success",
+      "131072", "Kill Failed", "Deny", "Failure",
+      "256", "Policy Disabled", "Allow", "Success",
+      "2097152", "Response Action Already Applied", "Deny", "Failure",
+      "4194304", "Response Failed", "Deny", "Failure"
+  ];
+  let parser = (starttime: datetime=datetime(null), 
+      endtime: datetime=datetime(null),
+      srcipaddr_has_any_prefix: dynamic=dynamic([]), 
+      dstipaddr_has_any_prefix: dynamic=dynamic([]), 
+      ipaddr_has_any_prefix: dynamic=dynamic([]),
+      dstportnumber: int=int(null), 
+      hostname_has_any: dynamic=dynamic([]), 
+      dvcaction: dynamic=dynamic([]), 
+      eventresult: string='*', 
+      disabled: bool=false) {
+      let src_or_any = set_union(srcipaddr_has_any_prefix, ipaddr_has_any_prefix); 
+      let dst_or_any = set_union(dstipaddr_has_any_prefix, ipaddr_has_any_prefix);
+      let alldata = CommonSecurityLog
+          | where not(disabled)
+              and (isnull(starttime) or TimeGenerated >= starttime)
+              and (isnull(endtime) or TimeGenerated <= endtime)
+          | where DeviceVendor == "CrowdStrike" and DeviceProduct == "FalconHost"
+          | where DeviceEventClassID in ("Network Access In A Detection Summary Event", "FirewallMatchEvent")
+          | where (array_length(hostname_has_any) == 0 or DestinationHostName has_any (hostname_has_any))
+              and (isnull(dstportnumber) or (DestinationPort == dstportnumber) or (AdditionalExtensions has tostring(dstportnumber)))
+      ;
+      let unparsedfirewalldata = alldata
+          | where DeviceEventClassID == "FirewallMatchEvent"
+          | parse-kv AdditionalExtensions as (deviceId: string, cmdLine: string, connectionDirection: int, eventType: string, hostName: string, icmpCode: int, icmpType: string, localAddress: string, localPort: int, matchCount: int, networkProfile: string, protocol: int, remoteAddress: string, remotePort: int, ruleAction: int, ruleDescription: string, ruleGroupName: string, ruleName: string, status: string) with (pair_delimiter=";", kv_delimiter="=")
+          | lookup EventFieldsLookup on ruleAction
+          | where ((array_length(dvcaction) == 0) or DvcAction has_any (dvcaction))
+          | where ((eventresult == "*") or EventResult == eventresult)
+          | extend
+              EventCount = matchCount,
+              EventStartTime = unixtime_milliseconds_todatetime(tolong(ReceiptTime)),
+              EventType = "NetworkSession";
```

Comment:
so we are getting two kind of logs from CrowdStrike, one Firewall and one from EndPoint. Are both of these collected from user's endpoint usin agent?
If Yes, why we are taking few logs as NetworkSession and few as EndpointNetworkSession?

EndpointNetworkSession - a network session reported by one of the end points of the session, including clients and servers. For such events, the schema supports the remote and local alias fields.

NetworkSession - a network session reported by an intermediate device monitoring the network, such as a Firewall, a router, or a network tap.

Reply:
Updated EventType for both as EndpointNetworkSession.